<div style="border:solid steelblue 3px; padding: 20px; border-radius: 10px">

**Привет, Александр!** 👋

Меня зовут Алексей Манелюк, я буду проверять твой проект. В Яндекс Практикуме принято общаться на "ты" 😉, но если некомфортно - скажи и перейдем на "Вы".

Моя цель - не указать на ошибки, а поделиться опытом и помочь в освоении анализа данных.

В тетрадке ты увидишь мои комментарии, пожалуйста, не удаляй, не перемещай и не изменяй их. У комментариев могут быть приписки **V2**, **V3**, **V...**, которые соответствуют каждой версии проверки. 
    
Я буду использовать цветовую разметку:
<p>
<div class="alert-danger" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ❌</b>

Так выделены критические замечания. Без их отработки я не смогу принять проект. </div>

<div class="alert-warning" style="padding: 20px; border-radius: 10px">
<b>  Комментарий ревьюера ⚠️</b>

Так выделены небольшие замечания или полезные советы. Постарайся учесть их.</div>
<p>
<div class="alert-success" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ✔️</b>

Так выделены правильные решения.</div>

Предлагаю работать в диалоге: если что-то дорабатываешь или отвечаешь на мой комментарий - пиши об этом. Мне будет удобней, если ты выделишь свои комментарии:
<p>
<div class="alert-info" style="padding: 20px; border-radius: 10px">
<b>Комментарий студента 💎</b>

Здесь исправил.</div>

Все это поможет мне быстрее принять твой проект.</div>

<div class="alert-info" style="padding: 20px; border-radius: 10px">
<b>Комментарий студента 💎</b>

Написать свой комментарий можно, скопировав этот блок в ячейку Markdown.</div>

# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



<div class="alert-warning" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ⚠️</b>

Несмотря на то, что первая часть этого проекта проверяется автоматизрованно, дам по ней пару советов - они пригодятся тебе в следующих проектах. 

**Сейчас их отрабатывать необязательно**

📌 После названия, обязательно нужно добавлять бриф, цели исследования, описание данных и план работы.</div>

## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


<div class="alert-warning" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ⚠️</b>

📌 Каждый раздел работы стоит завершать кратким выводом, о том что было сделано и какие результаты получены.

Как правильно оформить проект ты можешь прочитать в материалах курса:

`Полезные инструкции для учёбы - Оформление проекта - Рекомендации по выполнению проектов`
</div>

## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть. Тем более этот столбец не понадобится вам для исследования.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру. Проверьте остальные столбцы.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

Посмотрим сколько всего и какой процент людей в каждой группе по колличеству детей имеют долги.

In [22]:
gr_count = data.groupby('children')['debt'].count()
gr_sum = data.groupby('children')['debt'].sum()
percent_whith_debt = gr_sum/gr_count
gr = pd.merge(gr_count, gr_sum, right_index = True, left_index = True)
gr = pd.merge(gr, percent_whith_debt, right_index = True, left_index = True)
gr = gr.rename(columns={'debt_x':'count', 'debt_y':'sum', 'debt':'percent'})
gr

count   sum   percent
children                       
0         14091  1063  0.075438
1          4808   444  0.092346
2          2052   194  0.094542
3           330    27  0.081818
4            41     4  0.097561
5             9     0  0.000000

<div class="alert-success" style="border-radius: 10px; padding: 10px">
<b>Комментарий ревьюера V2 ✔️</b>

Это правильная сводная таблица со всеми нужными для сравнения категорий данными. 👌</div>

Больше всего долгов у людей без детей и с одним ребенком (1063 и 444 из 14091 и 4808 соответсвенно). <br>
 В семьях с двумя детьми 194 должника из 2052 человек. <br> В остальных группах в сумме 380 человек (что в 4 раза меньше, чем в самой наименьшей из трех крупнейших), из них 31 должник, что составляет 8.15% от общего числа. Группа слишком малочисленна, и процент в ней меньше, чем в других, поэтому ее можно не рассматривать. <br><br>
 Сравним процентное соотношение должников в трех крупнейших группах: в группах, где 1 или 2 ребенка примерно равные доли должников - 9.23% и 9.45%. У людей без детей процент людей с долгом 7.54%. Можно заметить разницу в более 1.5% между первой группой и оставшимися двумя, что является существенной разницей. Из этого можно сделать вывод, что группы людей с 1-2 детьми самые рискованные заемщики. 

**Вывод:** <br> 
- *Самые рисковые категории заемщиков: с 1 или 2 детьми* <br> 
- *Менее рисковые категории: все остальные*

<div class="alert-warning" style="border-radius: 10px; padding: 10px">
<b>Комментарий ревьюера V2 ⚠️</b>

Обрати внимание, что выборки несбалансированны, поэтому, если сравнить только самые большие, сопоставимые по размеру выборки с 0, 1 и 2 детьми, можно сказать что заемщики с большим количеством детей чаще допускают просрочку.</div>

<div class="alert-info" style="padding: 20px; border-radius: 10px">
<b>Комментарий студента 💎</b>

Переделал все пункты. Не получилось сделать сводную таблицу методом "groupby". Если не сложно, можешь, пожалуйста, показать, как это сделать в комментариях. </div>

<div class="alert-success" style="border-radius: 10px; padding: 10px">
<b>Комментарий ревьюера V2 ✔️</b>

С помощью `groupby()` нужную сводную таблицу можно вывести так: 👇</div>

In [27]:
# Код ревьюера
data.groupby('children')['debt'].agg(['count', 'sum', 'mean'])\
  .sort_values(by = 'mean', ascending = False)

count   sum      mean
children                       
4            41     4  0.097561
2          2052   194  0.094542
1          4808   444  0.092346
3           330    27  0.081818
0         14091  1063  0.075438
5             9     0  0.000000

<div class="alert-danger" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ❌</b>

Ты посчитал процент должников в каждой категории заемщиков. Но для сравнения нам также нужно общее количество заемщиков в каждой категории. 

Для наглядного сравнения нужно вывести сводную таблицу, в которой будут следующие столбцы:
- название категории
- количество заемщиков (кредитов)
- количество должников (этот столбец опциональный)
- средний процент просрочки по категории

Вывести такую таблицу можно с использованием методов как `groupby()`, так и `pivot_table()`.

Такие сводные таблицы нужно будет вывести для каждого вопроса, далее в работе. 

Когда ты выведешь такую таблицу, обрати внимание на то, что выборки не сбалансированы, поэтому выявлять зависимости стоит сравнивая самые многочисленные категории заемщиков.

Кроме того, для нашего заказчика - банка, разница даже в полпроцента на большом объеме кредитного портфеля будет существенна, поэтому здесь и далее давай выявим более и менее рискованные категории заемщиков.</div>

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [23]:

gr_count = data.groupby('family_status')['debt'].count()
gr_sum = data.groupby('family_status')['debt'].sum()
percent_whith_debt = gr_sum/gr_count
gr = pd.merge(gr_count, gr_sum, right_index = True, left_index = True)
gr = pd.merge(gr, percent_whith_debt, right_index = True, left_index = True)
gr = gr.rename(columns={'debt_x':'count', 'debt_y':'sum', 'debt':'percent'})
gr

count  sum   percent
family_status                              
Не женат / не замужем   2796  273  0.097639
в разводе               1189   84  0.070648
вдовец / вдова           951   63  0.066246
гражданский брак        4134  385  0.093130
женат / замужем        12261  927  0.075606

В самых малочисленных группах "в разводе" и "вдовец/вдова" самые маленькие доли должников 7.06% и 6.62% <br> В самой многочисленной группе "женат/замужем" 7.56% должников, что не сильно больше вышеупомянутых групп <br> 
Самый большой процент во второй и третьей группе по численности "гражданский брак" и "не женат/не замужем" 9.31% и 9.76% соответственно.

**Вывод:** <br> 
- *Самые рисковые категории заемщиков: "не женат/не замужем", "гражданский брак* <br> 
- *Менее рисковые категории: "в разводе", "вдовец/вдова", "женат/замужем"*

<div class="alert-success" style="border-radius: 10px; padding: 10px">
<b>Комментарий ревьюера V2 ✔️</b>

Верно, если обобщить категории по отношению к браку, можно сказать, что заемщики состоящие или состоявшие в браке выглядят более надежными.</div>

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [24]:
gr_count = data.groupby('total_income_category')['debt'].count()
gr_sum = data.groupby('total_income_category')['debt'].sum()
percent_whith_debt = gr_sum/gr_count
gr = pd.merge(gr_count, gr_sum, right_index = True, left_index = True)
gr = pd.merge(gr, percent_whith_debt, right_index = True, left_index = True)
gr = gr.rename(columns={'debt_x':'count', 'debt_y':'sum', 'debt':'percent'})
gr

count   sum   percent
total_income_category                       
A                         25     2  0.080000
B                       5014   354  0.070602
C                      15921  1353  0.084982
D                        349    21  0.060172
E                         22     2  0.090909

Группы "A", "D", "E" слишком малы по численности, чтобы объективно их оценить. <br>
Рассмотрим людей с доходами относящимися к "B" и "C":
- "В" - 7.06% должников
- "С" - 8.5% должников <br>
Заметим, что у людей с доходом, относящимся к категории "С", доля людей с долгом больше более чем на 1%, следовательно это самая рисковая группа людей.

**Вывод:** <br> 
- *Самые рисковые категории заемщиков: "С"* <br> 
- *Менее рисковые категории: "В"*

<div class="alert-success" style="border-radius: 10px; padding: 10px">
<b>Комментарий ревьюера V2 ✔️</b>

Верно, если сравнить самые большие категории B и C, можно сказать, что заемщики с более высоким доходом платят кредиты вовремя лучше.</div>

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [25]:
gr_count = data.groupby('purpose_category')['debt'].count()
gr_sum = data.groupby('purpose_category')['debt'].sum()
percent_whith_debt = gr_sum/gr_count
gr = pd.merge(gr_count, gr_sum, right_index = True, left_index = True)
gr = pd.merge(gr, percent_whith_debt, right_index = True, left_index = True)
gr = gr.rename(columns={'debt_x':'count', 'debt_y':'sum', 'debt':'percent'})
gr

count  sum   percent
purpose_category                              
операции с автомобилем     4279  400  0.093480
операции с недвижимостью  10751  780  0.072551
получение образования      3988  369  0.092528
проведение свадьбы         2313  183  0.079118

Все группы людей достаточно большие, чтоб объективно оценить их, сравнив доли должников. <br>
У 9.35% людей с кредитом на автомобиль есть долги - это наибольший показатель.   <br>
Затем идут люди с кредитом на образование. Их доля 9.25%, что не намного меньше самой рисковой группы. <br>
У 7.26% людей с кредитом на операции с недвижимостью есть долги - это наименьший показатель показатель. <br>
И 7.91% доля должников среди людей с кредитом на проведение свадьбы, что больше самой маленькой на около 0.6%, но меньше почти на 2% чем первые две группы.

**Вывод:** <br> 
- *Самые рисковые категории заемщиков: "операции с автомобилем", "получение образования"* <br> 
- *Менее рисковые категории: "операции с недвижимостью", "проведение свадьбы"*

<div class="alert-success" style="border-radius: 10px; padding: 10px">
<b>Комментарий ревьюера V2 ✔️</b>

Да, все верно! 👌</div>

<div class="alert-danger" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ❌</b>

Аналогичные замечания и по остальным вопросам по выявлению зависимостей просрочки от различных факторов.</div>

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* пропуски могут появлятся случайно, при внесеннии данных в ручную, из-за человеческого фактора. Человек специально может скрывать данные, либо их может не существовать вовсе. Также может быть техническая ошибка у аппарате используемом для измерений чего-либо.

<div class="alert-warning" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ⚠️</b>

Ты правильно указал человеческий фактор, как наиболее вероятную причину образования пропусков в данных. Но какая причина еще может вызвать потерю данных?</div>

<div class="alert-info" style="padding: 20px; border-radius: 10px">
<b>Комментарий студента 💎</b>

Дополнил.</div>

<div class="alert-success" style="border-radius: 10px; padding: 10px">
<b>Комментарий ревьюера V2 ✔️</b>

Согласен, техническую ошибку также исключать нельзя.</div>

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:*  среднее значение может искажать реальность, из-за выбросов (очень больших или маленьких значениий), которые сдвигают среднее ближе к своему значению.

<div class="alert-success" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ✔️</b>

Совершенно верно, медиана менее чувствительна к выбросам, чем среднее значение.</div>

### Шаг 4: общий вывод.

###### В этом исследоввнии я работал с датасетом, содержащим информацию о клиентах банка, которыевзяли кредит. В таблице находится такая информация, как количество детей, семейный статус, доход и цель кредита. Главной целью было выявить влияние этих факторов на возврат кредита в срок, чтобы определить надежность заемщиков. <br>
###### В исходных данных были пропуски и ошибки, которые были устранены на первом этапе работы. Далее для удобства данные были объединены в категории по доходу и по цели кредита.
###### Чтобы выявить определенные зависимости, в каждой группе были представленны сводные таблицы, состоящие из общего числа людей в группе, количества должников и их доли в общем числе. На их основании можно составить портреты идеального и самого рискового заемщика. 
- портрет идеального заемщика: человек без детей, находящийся в разводе или являющийся вдовцем/вдовой, имеющий доход от 200тыс до миллиона рублей, который берет кредит на преобретение недвижимости.
- портрет самого рискового заемщика: человек с одним или двумя детьми, холостой или находящийся в гражданском браке, с доходом от 50 до 200 тысяч рублей, который берет кредит на автомобиль. <br><br>
Подробные результаты исследования можно увидеть в самой работе.

<div class="alert-success" style="border-radius: 10px; padding: 10px">
<b>Комментарий ревьюера V2 ✔️</b>

Теперь у тебя получился подробный общий вывод, отражающий все основные результаты исследования. 👌</div>

<div class="alert-info" style="padding: 20px; border-radius: 10px">
<b>Комментарий студента 💎</b>

Переписал.</div>

<div class="alert-danger" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ❌</b>

Общий вывод - важный элемент работы. Его нужно сформулировать, взяв основные результаты из всех шагов проекта. Начать нужно с того, с какими данными мы работали и какие действия провели на этапе предобработки, сформулировать все выявленные зависимости просрочки от различных факторов (после внесения доработок), можно дать портрет идеального и рискованного заемщика и закончить выявленными при ответе на вопросы зависимостями и рекомендациями заказчику по улучшению сбора данных и системы скоринга.

Заказчик, зачастую, читает только общий вывод исследования - поэтому он должен быть максимально информативным.</div>

<div style="border:solid steelblue 3px; padding: 20px; border-radius: 10px">
    
<b>Общий комментарий ревьюера:</b>

Ты проделал большую работу, но нужно кое-что исправить, чтобы я ее принял.
    
Есть несколько критических замечаний:
- по выводу сводных таблиц для ответов по поиску зависимостей
- по выявлению зависимостей просрочки от различных факторов (дети, семейный статус, уровень дохода, цель кредита)
- по вопросу касательно заполнения пропусков медианой
- по общему выводу
    
Буду ждать твой доработанный проект. ⌛️</div>

In [26]:
print('Спасибо, Алексей, за проверку!')

Спасибо, Алексей, за проверку!


<div style="border:solid steelblue 3px; padding: 20px; border-radius: 10px">
    
<b>Общий комментарий ревьюера V2</b>

Тебе спасибо, что доработал проект! 🤝 Критических замечаний больше нет, получилась очень хорошая работа.
    
Успехов в дальнейшем обучении! 🏆
    
</div>